pyfifotax/utils.py

```diff
@@ -0,0 +1,584 @@
+    share_gains = sum_decimal(share_gain_series[share_gain_series > 0])
+
+    forex_gain_series = df_forex["Gain [EUR]"]
+
+    total_dividends = sum_decimal(df_dividends["Amount [EUR]"])
+    total_fees = sum_decimal(df_fees["Amount [EUR]"])
+    total_taxes = sum_decimal(df_taxes["Amount [EUR]"])
+
+    # unlike a previous version, we have to split things here
+    # losses from share can only be compared to gains from shares
+    # hence, there is a "total" gain/loss and then separate gains and
+    # losses from shares, e.g. one could have had a gain of 100
+    # and a loss of 150 over a year, i.e. a total loss of 50
+    total_foreign_gains = share_losses + share_gains + total_dividends
+    gains_from_shares = share_gains
+    losses_from_shares = -share_losses
+    total_gain_forex = sum_decimal(forex_gain_series)
+
+    anlagen = [
+        (
+            "Anlage KAP",
+            "Zeile 19: Ausländische Kapitalerträge (ohne Betrag lt. Zeile 47)",
+            round_decimal(total_foreign_gains, precision="0.01"),
+        ),
+        (
+            "Anlage KAP",
+            "Zeile 20: In den Zeilen 18 und 19 enthaltene Gewinne aus Aktienveräußerungen i. S. d. § 20 Abs. 2 Satz 1 Nr 1 EStG",
+            round_decimal(gains_from_shares, precision="0.01"),
+        ),
+        (
+            "Anlage KAP",
+            "Zeile 23: In den Zeilen 18 und 19 enthaltene Verluste aus der Veräuerung von Aktien i. S. d. § 20 Abs. 2 Satz 1 Nr. 1 EStG",
+            round_decimal(losses_from_shares, precision="0.01"),
+        ),
+        (
+            "Anlage KAP",
+            "Zeile 41: Anrechenbare noch nicht angerechnete ausländische Steuern",
+            round_decimal(total_taxes, precision="0.01"),
+        ),
+        (
+            "Anlage N",
+            "Zeile 48: (Werbungskosten Sonstiges): Überweisungsgebühren auf deutsches Konto für Gehaltsbestandteil RSU/ESPP",
+            round_decimal(total_fees, precision="0.01"),
+        ),
+        (
+            "Anlage SO",
+            "Zeilen 42 - 48: Gewinn / Verlust aus Verkauf von Fremdwährungen",
+            round_decimal(
+                total_gain_forex, precision="0.01"
+            ),  # here: the sum should be fine
+        ),
+    ]
+    summary = {
+        "ELSTER - Anlage": [a[0] for a in anlagen],
+        "ELSTER - Zeile (Suggestion!)": [a[1] for a in anlagen],
+        "Value": [a[2] for a in anlagen],
+    }
+    df_summary = pd.DataFrame(summary)
+    return df_summary
+
+
+def create_report_sheet(name: str, df: pd.DataFrame, writer: pd.ExcelWriter):
+    if "Date" in df:
+        df.sort_values("Date", inplace=True)
+    elif "date" in df:
+        df.sort_values("date", inplace=True)
+    elif "Sell Date" in df:
+        df.sort_values(["Sell Date", "Buy Date"], inplace=True)
+    elif "Meldezeitraum" in df:
+        df.sort_values("Meldezeitraum", inplace=True)
+    elif "ELSTER" in df.columns[0]:
```

Comment:
Get exception here: `IndexError('index 0 is out of bounds for axis 0 with size 0')`

Stack:
```
__getitem__, range.py:1020
create_report_sheet, utils.py:295
write_report_awv, utils.py:405
create_excel_report_awv, report_data.py:441
main, create_report.py:51
<module>, create_report.py:56
```

Reply:
Should disappear again if we don't create sheets for empty dataframes. I suppose that the AWV report is empty. Didn't consider that to be honest.
I'll revert to
- returning early for adding the "total" row and the create_sheet, the former allows the latter to work again as expected
- by not writing out empty dataframes to excel sheets, you shouldn't run into this issue in the first place